core/types/transaction.go

```diff
@@ -63,7 +63,7 @@ type txdata struct {
 	Hash *common.Hash `json:"hash" rlp:"-"`
 }
 
-type txdataMarshaling struct {
+type TxdataMarshaling struct {
```

Comment:
How come `TxdataMarshaling` needs to be made public? I don't see it being used outside of the `types` packages in this PR.

Reply:
Not needed, `txDataMarshalling` is only use a parameter to `gencoded` to modify marshalling rules... I'll change this